### PR TITLE
Fix bugs related to large system fonts

### PR DESCRIPTION
**Pull request by UrsKahmann**

### CODE REVIEW

File: app/src/main/kotlin/de/digitalService/useID/ui/components/pin/PinDigitField.kt [hunk 1]
```diff
@@ -22,10 +21,17 @@ import androidx.compose.ui.unit.dp
...
+    obfuscation: Boolean,
+    placeholder: Boolean,
+    modifier: Modifier = Modifier
+) {
     Box(
         modifier = modifier
-            .size(30.dp)
+            // To prevent the box from resizing after a first digit is entered, calulcate a suitable height
+            // with respect to the current system font scaling
+            .defaultMinSize(minWidth = 30.dp, minHeight = (15 + (LocalDensity.current.fontScale * 30)).dp)
```
Comment (FabioTacke) [inline app/src/main/kotlin/de/digitalService/useID/ui/components/pin/PinDigitField.kt:34]: So we do look up the font scaling for adjusting the UI after all? 🙃

**Review by FabioTacke — COMMENTED** [review]
Looks really good so far. I just noticed two screens where the layout seems to be a little off: On the home screen the button covers the image and on the number entry field the digits collide with the digit indicators at the bottom. I would like to ask you to take a look at those issues and see if we can resolve those remaining ones. 🙏

Screenshot_20230126_080903
Screenshot_20230126_074959

**Comment by UrsKahmann** [comment]
> Looks really good so far. I just noticed two screens where the layout seems to be a little off: On the home screen the button covers the image and on the number entry field the digits collide with the digit indicators at the bottom. I would like to ask you to take a look at those issues and see if we can resolve those remaining ones. 🙏
> 
> Screenshot_20230126_080903 Screenshot_20230126_074959

Thanks for pointing it out! I will take a look again. Regarding the image on the homescreen, I left it this way intentionally because I though it's not great but also not terrible and wasn't sure if I should invest more time into it 😅 

Regarding the digit field: What device/simulator did you test it on? Because this didn't happen on my side, neither on the pixel 3A sim nor on the galaxy S20FE real device 🤔  But I will have a look again an try to make it more robust :D

File: app/src/main/kotlin/de/digitalService/useID/ui/components/pin/PinDigitField.kt (hunk 1 above)
Comment (UrsKahmann) [reply app/src/main/kotlin/de/digitalService/useID/ui/components/pin/PinDigitField.kt:34]: Yes, I am not sure if this is a viable solution code wise, but I tried many things and wasn't able to find a way to make it work without taking the current scaling into account. Let me no if this is a no go and I will look into it again to find a different way :)

**Comment by FabioTacke** [comment]
Regarding the home screen you could also ask for the design teams's opinion. I'm fine with whatever they say. 😃
I'm using the Pixel 4 emulator.

File: app/src/main/kotlin/de/digitalService/useID/ui/components/pin/PinDigitField.kt (hunk 1 above)
Comment (FabioTacke) [reply app/src/main/kotlin/de/digitalService/useID/ui/components/pin/PinDigitField.kt:34]: No, it's fine. Doesn't hurt here. 👍

**Review by FabioTacke — APPROVED** [review]
Well done!

**Pull request merged**